torch/_dynamo/variables/lists.py

```diff
@@ -153,10 +154,28 @@ def call_method(
         elif name in cmp_name_to_op_mapping:
             left = self
             right = args[0]
-            if not isinstance(left, BaseListVariable) and not isinstance(
+            # TODO this type check logic mirrors the following
+            # https://github.com/python/cpython/blob/a1c52d1265c65bcf0d9edf87e143843ad54f9b8f/Objects/object.c#L991-L1007
+            # But we should probably move it up the stack to so that we don't
+            # need to duplicate it for different VTs.
+            if not isinstance(left, BaseListVariable) or not isinstance(
```

Comment:
This line is the fix for #148179.

The other stuff below make sure we return bools rather than `NotImplemented` (checked by the `conds` in `test_list_compare_polyfill_non_lists`)